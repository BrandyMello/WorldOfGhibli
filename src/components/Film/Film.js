import React from 'react';
import './Film.scss';
import PropTypes from 'prop-types';

const Film = ({ title, description, photo }) => {

  return (
      <article>
        <h2>{title}</h2>
        <img src={photo} className='movie-image' alt='movie-cover'></img>
        <p>{description}</p>
      </article>

  )
}

export default Film

Film.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
  photo: PropTypes.string
}
