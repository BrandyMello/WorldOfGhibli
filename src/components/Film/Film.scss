.movie-image {
  border-radius: 50%;
  height: 220px;
  overflow: hidden;
  position: relative;
  width: 220px;
}

.movie-container {
  display: inline-flex;
  margin-left: 8%;
  margin-top: 60px;
  z-index: 1;

  & h2,
  & p {
    background-color: #000000d1;
    padding: 15px;
  }

  & h2 {
    height: 40px;
    text-align: center;
  }

  & p {
    height: 600px;
    overflow: hidden;
    width: 200px;
  }
}