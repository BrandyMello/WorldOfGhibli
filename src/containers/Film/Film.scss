.movie-image {
  width: 220px;
  height: 220px;
  position: relative;
  overflow: hidden;
  border-radius: 50%;
}

.movie-container {
  display: inline-flex;
  margin-left: 7%;
  margin-top: 60px;
  z-index: 1;

  & h2,
  & p {
    background-color: #0000008a;
    padding: 15px;
  }

  & h2 {
    height: 40px;
    text-align: center;
  }

  & p {
    width: 200px;
    height: 600px;
    overflow: hidden;
  }
}