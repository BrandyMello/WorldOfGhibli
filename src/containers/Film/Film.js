import React from 'react';
import './Film.scss';

const Film = ({ title, description, photo }) => {

  return (
      <article>
        <h2>{title}</h2>
        <img src={photo} className='movie-image'></img>
        <p>{description}</p>
      </article>

  )
}

export default Film
