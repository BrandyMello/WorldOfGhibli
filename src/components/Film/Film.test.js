import React from 'react';
import Film from './Film';
import { shallow } from 'enzyme';

describe('Film', () => {

  it('should take a snapshot', () => {
    const wrapper = shallow(
      <Film
        key={12}
        title='Movie'
        description='Good'
        photo='imag.png'
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});