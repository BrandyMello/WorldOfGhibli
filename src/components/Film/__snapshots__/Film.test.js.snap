// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film should take a snapshot 1`] = `
<article>
  <h2>
    Movie
  </h2>
  <img
    className="movie-image"
    src="imag.png"
  />
  <p>
    Good
  </p>
</article>
`;
